src/bootstrap/service.rs

```diff
@@ -54,30 +58,35 @@ pub enum WorkUnitResponse {
     Finished = 5,
 }
 
-/// Request to commit a transaction batch to the database
+/// Response after a DB commit in a batch. If the DB is "full",
+/// we ask the threads to stop/start again
 enum DbCommitResponse {
     Continue,
     Stop,
 }
 
-struct DbCommitRequest {
-    txn_batch: RocksDbTransactionBatch,
-    num_messages: usize,
-    response_tx: oneshot::Sender<Result<DbCommitResponse, BootstrapError>>,
-}
-
-struct TrieInsertRequest {
-    trie_keys: Vec<Vec<u8>>,
-    response_tx: oneshot::Sender<Result<RocksDbTransactionBatch, BootstrapError>>,
-}
-
 #[derive(Clone)]
 struct WorkUnit {
     current_status: proto::ReplicationTriePartStatus,
     peer_address: String,
     shutdown_signal: Arc<AtomicBool>,
-    db_commit_tx: mpsc::Sender<DbCommitRequest>,
-    trie_insert_tx: mpsc::Sender<TrieInsertRequest>,
+    thread_engine: Arc<ShardEngine>,
+}
+
+// Shared data for all threads working on a shard. The DB and the Trie
+// are shared among all the vts.
+struct ShardSharedWorkData {
```

Comment:
Do we want all these things to be separately lockable? Or do we want 1 lock over the whole thing?